remerkleable/basic.py

```diff
@@ -73,21 +76,98 @@ def __new__(cls, value: int):
             raise ValueError(f"value out of bounds for {cls}")
         return super().__new__(cls, value)
 
-    def __add__(self, other):
+    def __add__(self: T, other: int) -> T:
         return self.__class__(super().__add__(self.__class__.coerce_view(other)))
 
-    def __sub__(self, other):
+    def __radd__(self: T, other: int) -> T:
+        return self.__add__(other)
+
+    def __sub__(self: T, other: int) -> T:
         return self.__class__(super().__sub__(self.__class__.coerce_view(other)))
 
-    def __mul__(self, other):
+    def __rsub__(self: T, other: int) -> T:
+        return self.__class__(self.__class__.coerce_view(other).__sub__(self))
+
+    def __mul__(self: T, other: int) -> T:
         return self.__class__(super().__mul__(self.__class__.coerce_view(other)))
 
-    def __floordiv__(self, other):  # Better known as "//"
+    def __rmul__(self: T, other: int) -> T:
+        return self.__mul__(other)
+
+    def __mod__(self: T, other: int) -> T:
+        return self.__class__(super().__mod__(self.__class__.coerce_view(other)))
+
+    def __rmod__(self: T, other: int) -> T:
+        return self.__class__(self.__class__.coerce_view(other).__mod__(self))
+
+    def __floordiv__(self: T, other: int) -> T:  # Better known as "//"
         return self.__class__(super().__floordiv__(self.__class__.coerce_view(other)))
 
-    def __truediv__(self, other):
+    def __rfloordiv__(self: T, other: int) -> T:
+        return self.__class__(self.__class__.coerce_view(other).__floordiv__(self))
+
+    def __truediv__(self: T, other: int) -> T:
         raise OperationNotSupported(f"non-integer division '{self} / {other}' "
-                                    f"is not valid for {self.__class__.type_repr()} type")
+                                    f"is not valid for {self.__class__.type_repr()} left hand type")
+
+    def __rtruediv__(self: T, other: int) -> T:
+        raise OperationNotSupported(f"non-integer division '{other} / {self}' "
+                                    f"is not valid for {self.__class__.type_repr()} right hand type")
+
+    def __pow__(self, other: int, modulo=None):
+        return self.__class__(super().__pow__(other, modulo))  # TODO: stricter argument checks?
+
+    def __rpow__(self, other, modulo=None):
```

Comment:
The same as `__pow__`
```suggestion
    def __rpow__(self: T, other, modulo=None) -> T:
```

Reply:
Thanks, I missed these, whoops.